docs/gen_reference_doc.py

```diff
@@ -35,690 +36,784 @@
-	return True
+    if desc.strip().startswith('hidden'):
+        return False
+    if internal:
+        return True
+    if desc.strip().startswith('internal'):
+        return False
+    return True
+
 
 def highlight_signature(s):
-	name = s.split('(', 1)
-	name2 = name[0].split(' ')
-	if len(name2[-1]) == 0: return s
+    name = s.split('(', 1)
+    name2 = name[0].split(' ')
+    if len(name2[-1]) == 0:
+        return s
+
+    # make the name of the function bold
+    name2[-1] = '**' + name2[-1] + '** '
 
-	# make the name of the function bold
-	name2[-1] = '**' + name2[-1] + '** '
+    # if there is a return value, make sure we preserve pointer types
+    if len(name2) > 1:
+        name2[0] = name2[0].replace('*', '\\*')
+    name[0] = ' '.join(name2)
 
-	# if there is a return value, make sure we preserve pointer types
-	if len(name2) > 1:
-		name2[0] = name2[0].replace('*', '\\*')
-	name[0] = ' '.join(name2)
+    # we have to escape asterisks, since this is rendered into
+    # a parsed literal in rst
+    name[1] = name[1].replace('*', '\\*')
 
-	# we have to escape asterisks, since this is rendered into
-	# a parsed literal in rst
-	name[1] = name[1].replace('*', '\\*')
+    # we also have to escape colons
+    name[1] = name[1].replace(':', '\\:')
 
-	# we also have to escape colons
-	name[1] = name[1].replace(':', '\\:')
+    # escape trailing underscores
+    name[1] = name[1].replace('_', '\\_')
 
-	# escape trailing underscores
-	name[1] = name[1].replace('_', '\\_')
+    # comments in signatures are italic
+    name[1] = name[1].replace('/\\*', '*/\\*')
+    name[1] = name[1].replace('\\*/', '\\*/*')
+    return '('.join(name)
 
-	# comments in signatures are italic
-	name[1] = name[1].replace('/\\*', '*/\\*')
-	name[1] = name[1].replace('\\*/', '\\*/*')
-	return '('.join(name)
 
 def html_sanitize(s):
-	ret = ''
-	for i in s:
-		if i == '<': ret += '&lt;'
-		elif i == '>': ret += '&gt;'
-		elif i == '&': ret += '&amp;'
-		else: ret += i
-	return ret
+    ret = ''
+    for i in s:
+        if i == '<':
+            ret += '&lt;'
+        elif i == '>':
+            ret += '&gt;'
+        elif i == '&':
+            ret += '&amp;'
+        else:
+            ret += i
+    return ret
+
 
 def looks_like_namespace(line):
-	line = line.strip()
-	if line.startswith('namespace'): return True
-	return False
+    line = line.strip()
+    if line.startswith('namespace'):
+        return True
+    return False
+
 
 def looks_like_blank(line):
-	line = line.split('//')[0]
-	line = line.replace('{', '')
-	line = line.replace('}', '')
-	line = line.replace('[', '')
-	line = line.replace(']', '')
-	line = line.replace(';', '')
-	line = line.strip()
-	return len(line) == 0
+    line = line.split('//')[0]
+    line = line.replace('{', '')
+    line = line.replace('}', '')
+    line = line.replace('[', '')
+    line = line.replace(']', '')
+    line = line.replace(';', '')
+    line = line.strip()
+    return len(line) == 0
+
 
```

Comment:
there are a few of these. does the formatting style require two blank lines between functions?

Reply:
Yes, note that this only occurs with top-level functions and would not apply to functions within a class definition.